src/NuGet.Clients/PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -549,7 +547,34 @@ private void SearchPackageInActivePackageSource(string searchText, bool useCache
                     ? Resx.Resources.Text_Loading
                     : string.Format(CultureInfo.CurrentCulture, Resx.Resources.Text_Searching, searchText);
 
-                _packageList.LoadItems(loader, loadingMessage, _uiLogger);
+                // start SearchAsync task for initial loading of packages
+                var searchResultTask = loader.SearchAsync(null, CancellationToken.None);
```

Comment:
Discussed offline, In case of UseCache true, it only uses Model cache to return results if cache exists but that still goes through loader and specific feed. 

I'll add more comments here for clarity.